src/components/AdditionalFeature.js

```diff
@@ -1,13 +1,23 @@
-import React from 'react';
+import React from "react";
+import { connect } from "react-redux";
+import { addFeature } from "../actions/carActions";
 
-const AdditionalFeature = props => {
+const AdditionalFeature = (props) => {
+  const handleClick = () => {
+    props.addFeature(props.feature);
+  };
   return (
     <li>
       {/* Add an onClick that will let you add a feature to your car */}
-      <button className="button">Add</button>
+      <button onClick={handleClick} className="button">
+        Add
+      </button>
       {props.feature.name} (+{props.feature.price})
     </li>
   );
 };
 
-export default AdditionalFeature;
+const mapDispatchToProps = {
+  addFeature,
+};
+export default connect(mapDispatchToProps)(AdditionalFeature);
```

Comment:
daca nu ai nevoie de mapStateToProps atunci ii dai undefined in loc. Iar tu i-ai dat mapDispatchToProps in loc de mapstatetoprops de aici si eroarea pe care o primesti

Reply:
Ok, am inteles.